test/jsontest.cpp

```diff
@@ -0,0 +1,47 @@
+#include "../parser.hpp"
+#include <gtest/gtest.h>
+
+TEST(jsontest, wrong_input_test) {
+
+  Parser test;
+
+  ASSERT_THROW(test.fileparser("badmonster1.json"),std::invalid_argument);
+
+};
+
+TEST(jsontest, diff_test) {
+
+  Parser test;
+
+  Parser test2;
+
+  test.fileparser("monster1.json");
+
+  test2.fileparser("monster2.json");
+
+  ASSERT_NE(test.readstring("name"),test2.readstring("name"));
+
+  ASSERT_NE(test.readint("hp"),test2.readint("hp"));
+
+  ASSERT_NE(test.readint("dmg"),test2.readint("dmg"));
+
+};
+
+TEST(jsontest,fileparser) {
+
+  Parser test;
+
+  test.fileparser("monster1.json");
+
+  ASSERT_EQ(test.readstring("name"), "Igor");
+
+  ASSERT_EQ(test.readint("hp"), 200);
+
+  ASSERT_EQ(test.readint("dmg"), 40);
+
+};
```

Comment:
Az a fájl igazából rossz. Persze mondhatod, hogy a Te hibakezelési stratégiád a hibák csendben átlépése, de pl. a fájl hiányakor meg nem ezt tetted, így most az összkép elég zavaros. És persze mindenféle karakter, esetleg a teljes kulcs vagy érték hiányával nem is küzd meg a program.